readthedocs/gold/forms.py

```diff
@@ -57,21 +57,25 @@ def get_customer_kwargs(self):
 
     def get_subscription(self):
         customer = self.get_customer()
-        try:
-            # TODO get the first sub more intelligently
-            subscriptions = customer.subscriptions.all(limit=5)
+
+        # TODO get the first subscription more intelligently
+        subscriptions = customer.subscriptions.all(limit=5)
+        if subscriptions.data:
+            # Update an existing subscription - Stripe prorates by default
             subscription = subscriptions.data[0]
             subscription.plan = self.cleaned_data['level']
-            if 'stripe_token' in self.cleaned_data:
+            if 'stripe_token' in self.cleaned_data and self.cleaned_data['stripe_token']:
```

Comment:
The error was on this line specifically. In the case where the user is swapping their credit card, the `stripe_token` is present and the API call succeeds and everything is good. In the case where the user is just changing plans, `stripe_token` is `None`, the API call fails and the `except` clause is taken which adds a new subscription.

Because of that possibility of just having a failed API call result in more subscriptions, I rewrote that part of this function.